core/src/wallet/tezos/delegation/TezosLikeOriginatedAccount.cpp

```diff
@@ -131,14 +169,39 @@ namespace ledger {
 
                     static inline void update_balance(std::shared_ptr<api::Operation>& op, BigInt& sum) {
```

Comment:
This method is similar to the inner range-based loop code in `getBalance` method.
Can you put that code into a static method/plain old function to avoid redundant code ?

Reply:
Even though this is not part of the PR, I agree about the fact it’s the same code. More importantly: it’s duplicated logic. However, we cannot move that code out of the struct because of the agnostic code. However, I have moved the struct definition at the namespace level so that I can re-use it.